server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -1097,6 +1101,27 @@ public void insert(InsertRowsOfOneDevicePlan insertRowsOfOneDevicePlan)
     }
   }
 
+  @Override
+  public void insert(InsertRowsPlan plan) throws QueryProcessException {
+    boolean allSuccess = true;
+    for (int i = 0; i < plan.getInsertRowPlanList().size(); i++) {
+      if (plan.getResults().containsKey(i)) {
+        allSuccess = false;
+        continue;
+      }
+      try {
+        insert(plan.getInsertRowPlanList().get(i));
+        plan.getResults().put(i, RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS));
+      } catch (QueryProcessException e) {
+        plan.getResults().put(i, RpcUtils.getStatus(e.getErrorCode(), e.getMessage()));
+        allSuccess = false;
+      }
+      if (!allSuccess) {
```

Comment:
do we need to insert the left records while one record is failed? @qiaojialin

Reply:
Thanks for your review, in my opinion, I support inserting the remaining records in case of one record insertion failure. After all, we support partial insertion success. Moreover, this implementation is consistent with the previous semantics of composing each record into an InsertRowPlan.